app/screens/permalink/permalink.js

```diff
@@ -109,6 +109,10 @@ export default class Permalink extends PureComponent {
     }
 
     componentWillReceiveProps(nextProps) {
+        if (nextProps.channelId === '') {
+            this.handleClose();
+        }
```

Comment:
In this component, `channelId` (or channel) depends on `post` where post is null on delete.  As a result, channelId is passed as empty string.